analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -142,6 +142,8 @@ public ProgramState ResetOperations()
         public override string ToString() =>
             Equals(Empty) ? "Empty" + Environment.NewLine : SerializeExceptions() + SerializeSymbols() + SerializeOperations() + SerializeCaptures();
 
+        public static implicit operator ProgramState[](ProgramState from) => new[] { from };
```

Comment:
This allows returning a single state from the `PreProcess` method:
```cs
public override ProgramState[] PreProcess(SymbolicContext context) =>
    context.SetOperationConstraint(BoolConstraint.True);
```

Reply:
Nice!

Reply:
What about null handling? Single `ProgramState null` is semantically equal to `new[] { }` and not `new [] { null }`

Reply:
Or is it `new[] { ProgramState.Empty }`?

Reply:
Empty not. `null` denotes `no state to explore`. While `Empty` denotes a state to explore.

If we can't find good semantics, let's create static field `ProgramState.None`  as empty array.

Reply:
I don't think we should handle null here at all. This leads to hard to diagnose bugs:

```cs
public ProgramState[] Test()
{
  return null;               // returns null
  return (ProgramState)null; // returns new ProgramState[0]
}

```

https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgMwAERNAwjQN4U3t2GFsevkcDGAbQC6NAGbIA9mAByqADYKaAXhoxFCgNw9BwsZJkAVAJ4AHAKYZ5S1TQAUDAJQalO/nrrEAnPYAkAESGcpo0UADO6qGCICzBNsoqaq4KAL4BTu6eXr6BwaaW1qERUbaxzPnmVgmqyZrpmbo0qU1N1F4AbGFgZgpQAMZQCDRSlsgQCFLI+g5MwU5NfNlQ4g7BYZEpCx7Z7PgA7DQAgsjjJgB0AKI9CCYAPAwAfPaNO4IWCuEWTdkH6hYAdxmFWkYFSWQ4LXIqSAA===

Reply:
Let's remove the operator then

Reply:
Yes. That is probably the best. It felt a bit too "javascripty" to begin with.